herddb-core/src/main/java/herddb/sql/CalcitePlanner.java

```diff
@@ -341,6 +341,84 @@ public TranslatedQuery translate(String defaultTableSpace, String query, List<Ob
         }
     }
 
+    private TranslatedQuery calculateShowCreateTable(String query, List<Object> parameters) {
+        String items[] = { "SHOW","CREATE", "TABLE"};
+        if(Arrays.stream(items).allMatch(query::contains)) {
+            query = query.substring(Arrays.stream(items).collect(Collectors.joining(" ")).length()).trim();
+            String tableSpace = "herd";
+            String tableName;
+            if (query.contains(".")) {
+                String tokens[] = query.split("\\.");
+                tableSpace = tokens[0].trim();
+                tableName = tokens[1].trim();
+            } else {
+                tableName = query.trim();
+            }
+            try {
+                Table t = manager.getTableSpaceManager(tableSpace).getTableManager(tableName).getTable();
+                if (t != null) {
```

Comment:
You should also check if the table is bound to an uncommitted transaction. On this case you have to handle it has not existing if we are not inside the scope of the transaction that created the table

Reply:
Can you please elaborate on this?

Reply:
(sorry yesterday when I sent the first feedback I did not have my laptop but I was using my smartphone).


see TableManager#createdInTransaction field.

If the client is issuing this query:
`SHOW CREATE TABLE tspace1.foo`
and the TableManager for _tspace1.foo_ has createdInTransaction = 1234 this means that that table has been created during transaction 1234 but the transaction still has not been committed yet, so the table should not be "visible" to other transactions.

I am seeing now that CalcitePlanner#translate does not known anything about transactions.

So in this case I prefer to throw a TableNotExistsException in case of createdInTransaction > 0.
This way we won't let other transactions see the table if not committed.
The tradeoff here is that the transaction that is creating the table won't be able to use "SHOW CREATE TABLE", but this is not a big deal, because usually SHOW CREATE TABLE is used by operators in order to inspect the contents of a database.

Reply:
Thanks for the explanation. I will make the change.